Content.Server/Shuttles/Systems/ArrivalsSystem.cs

```diff
@@ -293,6 +299,20 @@ private void OnArrivalsDocked(EntityUid uid, ArrivalsShuttleComponent component,
             };
             _deviceNetworkSystem.QueuePacket(uid, null, payload, netComp.TransmitFrequency);
         }
+
+        // Moffstation - Start - Arrivals start fixes
+        // This is ensuring we are arriving at the station, and not arrivals
+        if (TryGetArrivals(out EntityUid arrivals) &&
+            args.MapUid != Transform(arrivals).MapUid)
+        {
+            // raise first arrivals event
+            if (component.FirstArrival)
+            {
+                RaiseLocalEvent(uid, new FirstArrivalEvent());
+                component.FirstArrival = false;
+            }
+        }
```

Comment:
`if`s can be combined
```csharp 
if (A)
  if (B)
    dothing()

// same as
if (A && B)
  dothing()
```

And probably a cheap optimization is to make the `component.FirstArrival` first to avoid looking up the arrivals entity when it's not needed.